secant/Utils/Logging/LogStore.swift

```diff
@@ -0,0 +1,55 @@
+//
+//  LogStore.swift
+//  secant-testnet
+//
+//  Created by Lukáš Korba on 23.01.2023.
+//
+
+import Foundation
+import OSLog
+
+extension Date {
+    func timestamp() -> String {
+        let dateFMT = DateFormatter()
+        dateFMT.locale = Locale(identifier: "en_US_POSIX")
+        dateFMT.dateFormat = "yyyy/MM/dd HH:mm:ss.SSSS"
+        
+        return String(format: "%@", dateFMT.string(from: self))
+    }
+}
+    
+enum LogStore {
+    static func exportCategory(
+        _ category: String,
+        hoursToThePast: TimeInterval = 168,
+        fileSize: Int = 1_000_000
+    ) -> [String]? {
+        guard let bundle = Bundle.main.bundleIdentifier else { return nil }
+        
+        do {
+            let store = try OSLogStore(scope: .currentProcessIdentifier)
+            let date = Date.now.addingTimeInterval(-hoursToThePast * 3600)
+            let position = store.position(date: date)
+            var entries: [String] = []
+            var size = 0
+            
+            entries = try store
+                .getEntries(at: position)
+                .reversed()
+                .compactMap { $0 as? OSLogEntryLog }
+                .filter { entry -> Bool in
```

Comment:
I am not sure if the filter is the best approach here. Filter will go through all the logs no matter if you already reach max size or not. Maybe for loop which breaks when you reach max size would be better here. Logically both works the same. For loop probably saves some memory.

Reply:
I agree. A reduce function that's equivalent to a for loop that does this would be hard to code and harder to read.

Reply:
changed